service/src/main/java/com/codeforcommunity/requester/MapRequester.java

```diff
@@ -41,85 +45,138 @@ public MapRequester(Vertx vertx) {
+                                "Async Request to ArcGIS failed: " + asyncResult.toString());
+                          }
+                        }));
+              });
+        });
+  }
+
+  /** Make a request to update the ArcGIS feature layer */
+  private Future<Void> updateLayers(
+      List<String> blockIds, BlockStatus updateTo, Future<String> tokenFuture) {
+    logger.info("Making request to update blocks to " + updateTo.name());
+    return tokenFuture.compose(
+        tokenString ->
+            this.getBlockFidsFuture(blockIds, updateTo)
+                .compose(
+                    updateJson -> {
+                      MultiMap formData =
+                          MultiMap.caseInsensitiveMultiMap()
+                              .add("f", "json")
+                              .add("token", tokenString)
+                              .add("updates", updateJson.encode());
+
+                      return Future.future(
+                          promise -> {
+                            client
+                                .postAbs(featureLayerApplyEditsRoute)
+                                .putHeader("content-type", "multipart/form-data")
+                                .sendForm(
+                                    formData,
+                                    (ar) -> {
+                                      if (ar.succeeded()) {
+                                        HttpResponse<Buffer> httpResponse = ar.result();
 
-    return updateFuture;
+                                        if (httpResponse.statusCode() == 200) {
+                                          JsonObject responseBody = httpResponse.bodyAsJsonObject();
+                                          if (responseBody.containsKey("error")
+                                              && responseBody
+                                                      .getJsonObject("error")
+                                                      .getInteger("code")
+                                                  == 498) {
+                                            // The API token is invalid, reset it and make this call
+                                            // again.
+                                            logger.info(
+                                                "Remaking token request while updating blocks to "
+                                                    + updateTo.name());
+                                            this.tokenFuture = updateToken();
+                                            updateLayers(blockIds, updateTo, this.tokenFuture)
+                                                .onSuccess((V) -> promise.complete())
+                                                .onFailure(promise::fail);
+                                          } else if (responseBody.containsKey("updateResults")) {
+                                            // TODO: NOTE - THIS DOES NOT GUARANTEE A SUCCESS CASE
```

Comment:
This todo contradicts the comment below the log statement, make them the same comment